apps/ui/src/components/user/EditUserCard.vue

```diff
@@ -1,289 +1,307 @@
 const updateUserMutation = useMutation(UpdateUserDocument);
 const createUserGroupMutation = useMutation(CreateUserGroupDocument);
 const deleteUserGroupMutation = useMutation(DeleteUserGroupDocument);
+const createUserPermissionMutation = useMutation(CreateUserPermissionDocument);
+const deleteUserPermissionMutation = useMutation(DeleteUserPermissionDocument);
+const updateUserPermissionMutation = useMutation(UpdateUserPermissionDocument);
 
-sourceData.onResult(reloadInputData);
-onMounted(reloadInputData);
 
-const editor = ref<UserPermissionsEditor | null>(null);
-const submitting = ref<boolean>(false);
-const loadingError = ref<string>("");
-const inputUser = ref<Partial<User>>({});
-const userGroupsToBeDeleted = ref<UserGroup[]>([]);
-const groupsToBeAdded = ref<Pick<Group, "name" | "id">[]>([]);
-const userDetailsInput = ref<UserDetailsInput | null>(null);
-const showPermissionsEditor = ref<boolean>(false);
+sourceData.onResult((result) => {
+  if (result.loading) return;
+  const user = result.data.user;
 
-const confirmPermissionsEditor = computed({
-  get: () => showPermissionsEditor.value,
-  set: (value: boolean) => {
-    if (!value && editor.value.hasUnsavedChanges) {
-      dialog.warning({
-        title: "Close without saving?",
-        content: "Are you sure you want to close without saving?",
-        positiveText: "Yes",
-        negativeText: "Go Back",
-        onPositiveClick: () => {
-          showPermissionsEditor.value = false;
-        },
-      });
-    } else {
-      showPermissionsEditor.value = value;
-    }
-  },
-});
+  oldGroups.value = [];
+  newGroups.value = [];
+  oldPermissions.value = [];
+  newPermissions.value = [];
 
-const currentUserGroups = computed<UserGroup[]>(() => {
-  if (!sourceData.result.value?.user?.groups) {
-    return [];
+  userData.value = {
+    username: user?.username,
+    mail: user?.mail,
+    discord: user?.discord,
   }
-  return sourceData.result.value.user?.groups?.filter(
-    (ug) => !userGroupsToBeDeleted.value.find((ug2) => ug2.id === ug.id)
-  );
-});
 
-function reloadInputData() {
-  loadingError.value = "";
-  const result = sourceData.result.value;
-  if (!result?.user) {
-    loadingError.value = "User not found";
-  }
-  inputUser.value = { ...result?.user };
-  groupsToBeAdded.value = [];
-  userGroupsToBeDeleted.value = [];
-}
+  const groups = user?.groups;
+  if (groups)
+    for (let i = 0; i < groups.length; i++) {
+      oldGroups.value[i] = groups[i];
+      newGroups.value[i] = groups[i].group as Group;
+    }
+
+
+  const permissions = user?.permissions;
+  if (permissions)
+    for (let i = 0; i < permissions.length; i++) {
+      oldPermissions.value[i] = permissions[i];
+      newPermissions.value[i] = permissions[i];
+    }
+
 
-async function save() {
-  if (!userDetailsInput.value?.isValid) {
+  const personResult = user?.person;
+  if (personResult)
+    person.value = personResult;
```

Comment:
`person` is readonly when retrieved from the API, but this value is passed in to be modified by `PeopleRow`. Create a copy of the data (e.g. `structuredClone(person)`).

Reply:
In the end, it doesn't matter, props are readonly unless they're in some kind of object, like an array.

Reply:
I should've elaborated more: it was passing `person` into the `PeopleRow` component. The `PeopleRow` component modifies the id and name of the passed in object, but it resulted in an error because the object was read-only. Seems like you've added the `structuredClone(...)` call anyway though, so it looks good to me now 👍